static/js/compose.js

```diff
@@ -561,28 +584,9 @@ exports.validate_stream_message_address_info = function (stream_name) {
     if (stream_data.is_subscribed(stream_name)) {
         return true;
     }
-
-    var response;
-
-    switch (check_unsubscribed_stream_for_send(stream_name,
-                                               page_params.narrow_stream !== undefined)) {
-    case "does-not-exist":
-        response = "<p>The stream <b>" +
-            Handlebars.Utils.escapeExpression(stream_name) + "</b> does not exist.</p>" +
-            "<p>Manage your subscriptions <a href='#streams/all'>on your Streams page</a>.</p>";
-        compose_error(response, $('#stream'));
-        return false;
-    case "error":
-        compose_error(i18n.t("Error checking subscription"), $("#stream"));
-        return false;
-    case "not-subscribed":
-        response = "<p>You're not subscribed to the stream <b>" +
-            Handlebars.Utils.escapeExpression(stream_name) + "</b>.</p>" +
-            "<p>Manage your subscriptions <a href='#streams/all'>on your Streams page</a>.</p>";
-        compose_error(response, $('#stream'));
-        return false;
-    }
-    return true;
+    var autosubscribe = page_params.narrow_stream !== undefined;
+    var error_type = check_unsubscribed_stream_for_send(stream_name, autosubscribe);
+    return validation_error(error_type, stream_name);
```

Comment:
I think the function name here, `validation_error`, make you think the function doesn't have side effects, but it can cause `compose_error(..., $('stream')` as a side effect.

It might be cleaner to have the function just return a string, and then say something like this:

~~~ js
error_response = validation_error(error_type, stream_name)
if (error_response) {
    compose_error(error_response, $('#stream'));
}
return error_response

Reply:
(I would also be fine with changing the name to something like `maybe_report_validation_error`.

Reply:
Addressed this in a separate commit. The function is renamed to `report_validation_error` because after we get rid of the synchronous check in ddf694e, it causes `compose_error` every time it's called.